src/server/routes/register.js

```diff
@@ -4,22 +4,36 @@ const UserModel = require('../models/user')
 const logger = require('../logger.js')
 const Utilities = require('../utilities')
 
-const saltSize = 70
+const SALT_SIZE = 70
 
 router.post('/', (req, res) => {
-
-	UserModel.on('index', err => logger.error(`Not created indexes: ${err}`))
 	
-	const userInstance = new UserModel(req.body)
+	const userData = {
+		username: req.body.username,
+		email: req.body.email,
+		password: req.body.password,
+		salt: ''
+	}
 
-	userInstance.salt = Utilities.generateSalt(saltSize)
-	userInstance.password = Utilities.hashPassword(userInstance.salt, userInstance.password)
-	userInstance.save()
-		.then( () => {return res.status(200).send('Created new user')})
-		.catch( (err) => {
+	const userInstance = new UserModel(userData)
+	
+	userInstance.validate(err => {
+		if(err) {
 			logger.error(err)
 			return res.status(500).send(err.message)
-		})
+		}
+
+		userInstance.salt = Utilities.generateSalt(SALT_SIZE)
+		userInstance.password = Utilities.createSaltedHash(userInstance.salt, userInstance.password)
```

Comment:
Can we set it like a property? Do docs allow for that?

Reply:
This is correct. userInstance is like mongo document.

Reply:
But you do it after `.validate` so `password` and `salt` are unvalidated.